### PR TITLE
Add a failover logic when an RPC is unavailable

### PR DESCRIPTION
**Pull request by cubismic**
I added a failover logic when an RPC is unavailable.
Now the indexer doesn't stop when an RPC is unavailable.

This resolves https://github.com/CodeChain-io/codechain-indexer/issues/118

### CODE REVIEW

File: src/worker/index.ts [hunk 1]
```diff
@@ -64,10 +75,19 @@ export default class Worker {
                 console.log("================ sync done ===================\n");
             })
             .catch(err => {
+                const error = err as Error;
+                console.log(JSON.stringify(error, null, 2));
```
Comment (cubismic) [inline src/worker/index.ts]: Remove log

File: src/routers/asset.ts [hunk 2]
```diff
@@ -56,12 +57,16 @@ import {
...
+    const syncIfNeeded: RequestHandler = async (req, res, next) => {
         if (req.query.sync === true) {
             try {
                 await context.worker.sync();
-            } catch (error) {
-                next(error);
+            } catch (e) {
+                const error = e as Error;
+                if (error.message.search(/ECONNRESET|ECONNREFUSED/) >= 0) {
+                    res.status(SERVICE_UNAVAILABLE).send();
```
Comment (joojis) [inline src/routers/asset.ts]: return

File: src/worker/index.ts [hunk 3]
```diff
@@ -43,17 +43,26 @@ export default class Worker {
...
         });
         this.watchJob.invoke();
     };
 
     public sync = async () => {
         const { sdk } = this.context;
-        const chainBestBlockNumber = await sdk.rpc.chain.getBestBlockNumber();
+        const chainBestBlockNumber = await sdk.rpc.chain
+            .getBestBlockNumber()
+            .catch(err => {
```
Comment (joojis) [inline src/worker/index.ts]: Please remove it.

File: test/api.spec.ts [hunk 4]
```diff
@@ -20,14 +18,54 @@ afterAll(async done => {
...
-    request(app)
-        .get("/api/ping")
-        .expect(200)
-        .end(err => {
-            if (err) {
-                throw err;
-            }
-            done();
-        });
+describe("API", () => {
```
Comment (joojis) [inline test/api.spec.ts]: For consistency, let's just use `test` here.

**Pull request merged**